File: 875-longest-mountain-in-array/longest-mountain-in-array.java
class Solution {
        public int longestMountain(int[] A) {
        int max = 0, i = 1;
        
        while (i < A.length) {
            int up = 0, down = 0;
            
            while (i < A.length && A[i - 1] < A[i]) { i++; up++;   }
            while (i < A.length && A[i - 1] > A[i]) { i++; down++; }
            
            if (up > 0 && down > 0) max = Math.max(max, up + down + 1);
            
            while (i < A.length && A[i - 1] == A[i]) i++;
        }
        return max;
    }
}

File: 875-longest-mountain-in-array/longest-mountain-in-array.py
class Solution(object):
    def longestMountain(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        if len(arr) < 3:
            return 0
        ret=0
        for i in range(1,len(arr)-1):
            if arr[i-1]<arr[i]>arr[i+1]:
                l=r=i
                while l>0 and arr[l]>arr[l-1]:
                    l-=1
                while r<len(arr)-1 and arr[r]>arr[r+1]:
                    r+=1
                ret =max(ret,r-l+1)
        return ret